packages/webapp/src/components/ui/input/SecretInput.tsx

```diff
@@ -13,13 +13,17 @@ type SecretInputProps = Omit<JSX.IntrinsicElements['input'], 'defaultValue'> & {
     refresh?: () => void;
 };
 
-const SecretInput = forwardRef<HTMLInputElement, SecretInputProps>(function PasswordField({ className, copy, ...props }, ref) {
+const SecretInput = forwardRef<HTMLInputElement, SecretInputProps>(function PasswordField(
+    { className, copy, optionalvalue, setoptionalvalue, defaultValue, refresh, ...props },
+    ref
+) {
     const [isSecretVisible, setIsSecretVisible] = useState(false);
 
-    const [changedValue, setChangedValue] = useState(props.defaultValue);
+    const [changedValue, setChangedValue] = useState(defaultValue);
 
-    const value = props.optionalvalue === null ? '' : props.optionalvalue || changedValue;
-    const updateValue = props.setoptionalvalue || setChangedValue;
+    // Handle null value in optionalvalue
+    const value = optionalvalue ?? changedValue;
```

Comment:
SecretInputProps never define `null` as accepted value, I wonder where it was happening and if it's allowed can you modify this type? (and maybe the other type too)

Reply:
Just updated the data types for these. Thanks @bodinsamuel for this.

Reply:
I still can't pass `optionalValue={null}` => `Type 'null' is not assignable to type 'string | undefined'.`

Reply:
Should now work ☺️.Thanks